### PR TITLE
Retail search samples with tests (new)

### PR DESCRIPTION
**Pull request by vasilypascal**
* Added retails search samples
* Added tests
* Created Runner project, which executes the sample by it's name with the help of ExampleAttributeHelper class

### CODE REVIEW

File: retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithFilteringTest.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        [Fact]
+        public void TestSearchWithFiltering()
+        {
+            const string ExpectedProductTitleName = "Tee";
+            const string ExpectedProductColor = "Black";
+
+            var firstPage = SearchWithFilteringTutorial.Search().First();
+
+            Assert.Contains(firstPage, result => result.Product.Title.Contains(ExpectedProductTitleName) && result.Product.Title.Contains(ExpectedProductColor));
+            Assert.Contains(firstPage, result => result.Product.ColorInfo.ColorFamilies.Contains(ExpectedProductColor));
```
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithFilteringTest.cs:31]: The check is not redundant.
In first Assert we verify that the product's **productTitle** consists of words "Tee" and "Black", in second verify we also check that the product's **colorFamilies** contains expected color - "Black"

File: retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithOrderingTest.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public class SearchWithOrderingTest
+    {
+        [Fact]
+        public void TestSearchWithOrdering()
+        {
+            const int ExpectedProductPrice = 39;
+
+            var firstPage = SearchWithOrderingTutorial.Search().First();
+
+            Assert.Equal(ExpectedProductPrice, firstPage.Results[0].Product.PriceInfo.Price);
```
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithOrderingTest.cs:29]: That really makes sense. I will improve

File: retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithQueryExpansionTest.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        [Fact]
+        public void TestSearchWithQueryExpansion()
+        {
+            const string ExpectedProductTitle = "Google Youth Hero Tee Grey";
+
+            var searchResultPages = SearchWithQueryExpansionTutorial.Search();
+
+            var firstPage = searchResultPages.ToArray()[0];
+            var thirdPage = searchResultPages.ToArray()[2];
```
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithQueryExpansionTest.cs:30]: Agree.  Will improve that

File: retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpecSample.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    private SearchRequest GetSearchRequest(string query, string facetKeyParam, string projectNumber)
+    {
+        string defaultSearchPlacement = $"projects/{projectNumber}/locations/global/catalogs/default_catalog/placements/default_search";
+
+        var facetKey = new SearchRequest.Types.FacetSpec.Types.FacetKey
+        {
+            Key = facetKeyParam
+        };
+
+        // Put the intervals here:
```
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpecSample.cs:37]: Agree. Added explanation for the "Intervals" below

File: retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithOrderingTest.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithOrderingTest.cs]: Resolved.

File: retail/interactive-tutorial/RetailSearch.Samples/SearchWithPaginationSample.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            Console.WriteLine($"TotalSize: {firstPage.TotalSize},");
+            Console.WriteLine("Items found in first page:");
+
+            foreach (SearchResponse.Types.SearchResult item in firstPage)
+            {
+                Console.WriteLine(item);
+            }
+        }
+
+        // Paste call with next page token here:
```
[the remark this answers is not in the document]
Comment (vasilypascal) [reply retail/interactive-tutorial/RetailSearch.Samples/SearchWithPaginationSample.cs]: Resolved.

File: retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithFilteringTest.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithFilteringTest.cs:31]: Acked!

**Review by amanda-tarafa — APPROVED** [review]
LGTM!

Thanks!

**Comment by amanda-tarafa** [comment]
Failures in CI are unrelated and being tracked in #1623.

**Pull request merged**